Rock/Communication/Medium/Sms.cs

```diff
@@ -179,38 +197,92 @@ private void CreateCommunication( int fromPersonAliasId, string fromPersonName,
             Rock.Transactions.RockQueue.TransactionQueue.Enqueue( transaction );
         }
 
+        /// <summary>
+        /// Generate a randomized list of available response codes that can be used for SMS tracking.
+        /// </summary>
+        /// <param name="rockContext">A context to use for database calls.</param>
+        /// <returns>A randomized <see cref="List{T}"/> of strings that are available for use.</returns>
+        static private List<string> GenerateAvailableResponseCodeList( Rock.Data.RockContext rockContext )
+        {
+            DateTime tokenStartDate = RockDateTime.Now.Subtract( new TimeSpan( TOKEN_REUSE_DURATION, 0, 0, 0 ) );
+            int[] blacklist = new int[] { 666, 911 };
+            int chunkSize = 100;
+
+            //
+            // Generate a list of codes that are currently active in the database.
+            //
+            var activeCodes = new CommunicationRecipientService( rockContext ).Queryable()
+                                    .Where( c => c.ResponseCode.StartsWith( "@" ) && c.CreatedDateTime > tokenStartDate )
+                                    .Select( c => c.ResponseCode )
+                                    .ToList();
+
+            //
+            // Starting at code 100, try to generate a list of available codes in small chunks until
+            // we have a list with atleast 1 available code.
+            //
+            for ( int startValue = 100; startValue < RESPONSE_CODE_MAX - chunkSize; startValue += chunkSize )
+            {
+                var availableCodes = Enumerable.Range( startValue, chunkSize )
+                    .Where( i => !blacklist.Contains( i ) )
+                    .Select( i => string.Format( "@{0}", i ) )
+                    .Where( c => !activeCodes.Contains( c ) )
+                    .ToList();
+
+                if ( availableCodes.Any() )
+                {
+                    availableCodes.OrderBy( c => Guid.NewGuid() ).ToList();
+                    return availableCodes;
+                }
```

Comment:
@cabal95 The orderby actually works, but the reordered list is not saved to a new list. If we just change these two lines to...
`return availableCodes.OrderBy( c => Guid.NewGuid() ).ToList();`
it works fine. I'll just pull this in and make that change.

Reply:
Oh, that will do it. I must have had that in two lines when I tested and then refactored it into two right before committing without retesting. Sorry 'bout that!

Reply:
@cabal95 No Prob!  Thanks for doing all these pull requests. It's awesome!